Src/AutoFixture/Kernel/TypeEnvy.cs

```diff
@@ -11,17 +11,12 @@ internal static class TypeEnvy
         public static TAttribute GetAttribute<TAttribute>(object candidate)
             where TAttribute : Attribute
         {
-            var member = candidate as MemberInfo;
-            if(member != null)
-            {
-                return member.GetCustomAttribute<TAttribute>();
-            }
-            var parameter = candidate as ParameterInfo;
-            if(parameter != null)
-            {
-                return parameter.GetCustomAttribute<TAttribute>();
-            }
-            return null;
+            var attributeProvider = candidate as ICustomAttributeProvider;
+
+            return attributeProvider?
+                .GetCustomAttributes(typeof(TAttribute), true)
+                .Cast<TAttribute>()
```

Comment:
Well, I could rewrite this to avoid LINQ usage, but currently I don't see how this could affect performance, given that we don't have any closures here. Do you have some particular issue in mind?

P.S. We have also a similar usage [here](https://github.com/AutoFixture/AutoFixture/blob/96fab6bfd5c5e6b4967ff92cbf04e9a7c96c5e91/Src/AutoNSubstitute/SubstituteAttributeRelay.cs#L31-L40) and no issues were reported so far.